src/Pact/Analyze/Types/Languages.hs

```diff
@@ -160,6 +167,23 @@ data Core t a where
   -- for NOT, and two operands for AND or OR.
   Logical :: LogicalOp -> [t Bool] -> Core t Bool
 
+instance (UserShow a, UserShow (t a), UserShow (t Integer), UserShow (t Decimal), UserShow (t Time))
+  => UserShow (Core t a) where
+  userShowsPrec d = \case
+    Lit a                     -> userShowsPrec d a
+    Sym s                     -> tShow s
+    StrConcat x y             -> parenList ["+", userShow x, userShow y]
```

Comment:
@bts I would like to convert all of these to, eg, `parenList [SAddition, userShow x, userShow y]`. Unfortunately this induces a circular import:

```
Module imports form a cycle:
         module `Pact.Analyze.Feature' (/home/joel/code/github/kadena
-io/pact/src/Pact/Analyze/Feature.hs)
        imports `Pact.Analyze.Types' (/home/joel/code/github/kadena-i
o/pact/src/Pact/Analyze/Types.hs)
  which imports `Pact.Analyze.Types.Languages' (/home/joel/code/githu
b/kadena-io/pact/src/Pact/Analyze/Types/Languages.hs)
  which imports `Pact.Analyze.Feature' (/home/joel/code/github/kadena
-io/pact/src/Pact/Analyze/Feature.hs)
```

It's not clear how I should avoid this without moving all instances into `Feature`. What do you think?

Reply:
@bts I would like to convert all of these to, eg, parenList [SAddition, userShow x, userShow y]. Unfortunately this induces a circular import:

```
Module imports form a cycle:
         module `Pact.Analyze.Feature' (/home/joel/code/github/kadena
-io/pact/src/Pact/Analyze/Feature.hs)
        imports `Pact.Analyze.Types' (/home/joel/code/github/kadena-i
o/pact/src/Pact/Analyze/Types.hs)
  which imports `Pact.Analyze.Types.Languages' (/home/joel/code/githu
b/kadena-io/pact/src/Pact/Analyze/Types/Languages.hs)
  which imports `Pact.Analyze.Feature' (/home/joel/code/github/kadena
-io/pact/src/Pact/Analyze/Feature.hs)
```

It's not clear how I should avoid this without moving all instances into Feature. What do you think?